inc/integrations/class-google-tag-manager.php

```diff
@@ -65,4 +70,29 @@ public function render_container_id_input() {
 			<input type="text" name="irving_integrations[<?php echo esc_attr( 'gtm_container_id' ); ?>]" value="<?php echo esc_attr( $gtm_key ); ?>" />
 		<?php
 	}
+
+	/**
+	 * Get data layer info for a page load.
+	 *
+	 * @param array $options Array of integration options.
+	 * @return array An array of options.
+	 */
+	public function get_data_layer( $options ) {
+		/**
+		 * Filters the data layer provided to GTM for each page.
+		 *
+		 * @param array $data_layer The data layer arguments for GTM.
+		 */
+		$data_layer = apply_filters(
+			'wp_irving_gtm_data',
+			[
+				'title' => wp_title( null, false ),
```

Comment:
I think this is slightly redundant with `document.title` but no real argument from me against passing it. I am wondering though if these should be namespaced under `irving`, as `title` would be a very common top-level data layer key.

Reply:
You're right that it's redundant, but I've seen occasions where the event fires before document.title is updated on the front end, so this ensures that we're always getting the right title from WP.